extensions/VisualEditor/wikia/modules/ve/ui/tools/ve.ui.WikiaDialogTool.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * UserInterface WikiaSourceModeDialog tool.
+ *
+ * @class
+ * @extends ve.ui.IconTextButtonTool
+ *
+ * @constructor
+ * @param {OO.ui.ToolGroup} toolGroup
+ * @param {Object} [config] Config options
+ */
+ve.ui.WikiaSourceModeDialogTool = function VeUiWikiaSourceModeDialogTool( toolGroup, config ) {
+	// Parent constructor
+	ve.ui.WikiaSourceModeDialogTool.super.call( this, toolGroup, config );
+};
+
+OO.inheritClass( ve.ui.WikiaSourceModeDialogTool, ve.ui.DialogTool );
+
+ve.ui.WikiaSourceModeDialogTool.static.name = 'wikiaSourceMode';
+ve.ui.WikiaSourceModeDialogTool.static.group = 'object';
+ve.ui.WikiaSourceModeDialogTool.static.icon = 'source';
+ve.ui.WikiaSourceModeDialogTool.static.title = 'Source Mode';
```

Comment:
Is there something stopping you from start using OO.ui.deferMsg? If so then please put review comment here so we don't end up going on production with hardcoded string.